src/modules/scraper/adapter/messaging/DepositReferralConsumer.ts

```diff
@@ -46,7 +47,27 @@ export class DepositReferralConsumer {
         if (nonce === 0) referralAddress = undefined;
       }
     }
-    await this.depositRepository.update({ id: deposit.id }, { referralAddress: referralAddress || null });
+
+    if (!referralAddress) {
+      // if no referral address is found, then look for the previous referral address used by this account
+      const previousReferralDeposit = await this.depositRepository.findOne({
+        where: {
+          depositorAddr: deposit.depositorAddr,
+          depositDate: LessThanOrEqual(deposit.depositDate),
+          stickyReferralAddress: Not(IsNull()),
+        },
+        order: { depositDate: { direction: "DESC", nulls: "LAST" } },
+      });
+      stickyReferralAddress = previousReferralDeposit?.stickyReferralAddress;
+    }
+
+    await this.depositRepository.update(
+      { id: deposit.id },
+      {
+        referralAddress: referralAddress || null,
+        stickyReferralAddress: stickyReferralAddress || referralAddress || null,
```

Comment:
store referral address and sticky referral address as separate fields, so we never lose the on-chain data